tang.go

```diff
@@ -71,31 +118,35 @@ func (p TangPin) decrypt(msg *jwe.Message) ([]byte, error) {
 		return nil, fmt.Errorf("cannot parse provided token, node 'clevis.tang.url'")
 	}
 
-	headers := msg.Recipients()[0].Headers()
+	headers := msg.ProtectedHeaders()
 
 	receivedKey, err := performEcmrExhange(p.URL, advertizedKeys, headers.KeyID(), headers.EphemeralPublicKey())
 	if err != nil {
 		return nil, err
 	}
 
-	if err := headers.Set(jwe.AlgorithmKey, jwa.ECDH_ES); err != nil {
-		return nil, err
-	}
-	newEpk, err := jwk.New(receivedKey)
+	keysize, err := keySize(headers.ContentEncryption())
 	if err != nil {
 		return nil, err
 	}
-	if err := headers.Set(jwe.EphemeralPublicKeyKey, newEpk); err != nil {
+
+	zBytes := expandBuffer(receivedKey.X.Bytes(), roundup(receivedKey.Curve.Params().BitSize, 8)/8)
```

Comment:
It's simpler to do `(BitSize + 7) / 8`.  

And since it looks like `roundup` is only used here, why not rename it `roundToBytes` and just pass in `BitSize`?

Or even introduce a
```
func expandBufferWithBits(b []buffer, s int) int {
    return expandBuffer(b, (s + 7) / 8)
}
```

Reply:
added function `divRoundUp()` that might be reused for other cases